avm/res/web/site/tests/e2e/webConfig/main.test.bicep

```diff
@@ -0,0 +1,79 @@
+targetScope = 'subscription'
+
+metadata name = 'Web App'
+metadata description = 'This instance deploys the module as Web App with the set of api management configuration.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-web.sites-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'wswc'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    serverFarmName: 'dep-${namePrefix}-sf-${serviceShort}'
+    apiManagementName: 'dep-${namePrefix}-apim-${serviceShort}'
+    applicationInsightsName: 'dep-${namePrefix}-appInsights-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      kind: 'app'
+      serverFarmResourceId: nestedDependencies.outputs.serverFarmResourceId
+      appInsightResourceId: nestedDependencies.outputs.applicationInsigtsId
+      apiManagementConfiguration: {
+        id: '${nestedDependencies.outputs.apiManagementId}/apis/todo-api'
+      }
+      managedIdentities: {
+        systemAssigned: true
+      }
+      siteConfig: {
+        alwaysOn: true
+        appCommandLine: ''
+      }
+      appSettingsKeyValuePairs: {
+        SCM_DO_BUILD_DURING_DEPLOYMENT: 'False'
+        ENABLE_ORYX_BUILD: 'True'
+      }
+    }
+    dependsOn: [
+      nestedDependencies
+    ]
```

Comment:
```suggestion
```

Reply:
I don't quite understand why `dependsOn` isn't needed here, as I'm seeing the same usage in other test files as well. Are they any different?

Reply:
Hey @zedy-wj,
fair question. There is an [issue](https://github.com/Azure/bicep/issues/13457) that I created a while ago, arguing that sometimes ARM/Bicep doesn't seem to wait for dependencies consistently. Back in the day, we did add explicit `dependsOn` all over as it 'seemed' to help with the issue. Later it turned out that this was actually not the case and it's a backend topic (caching to be precise). So whenever I see it now, I request the removal of the explicit `dependsOn`, as the implicit dependency will work just as well :)

Reply:
Got it, I've updated. Actually, I think implicit dependency is enough, but I'm curious why some tests still use `dependsOn`. Thanks for your answer!

Reply:
Just a lack of cleanup that's all. We never bothered to invest too much time on the dependsOn removal as it doesn't hurt. It's just redundant 😉